openstates/dc/bills.py

```diff
@@ -1,212 +1,449 @@
-        url = 'http://dcclims1.dccouncil.us/lims/print/list.aspx?FullPage=True&Period=' + session
-
-        html = self.urlopen(url)
-        doc = lxml.html.fromstring(html)
-        rows = doc.xpath('//table/tr')
-        # skip first row
-        for row in rows[1:]:
-            bill_id, title, _ = row.xpath('td/text()')
-            title = title.replace(u'\xe2\x80\x99', "'")  # smart apostrophe
-
-            if bill_id.startswith('B'):
-                type = 'bill'
-            elif bill_id.startswith('CER'):
-                type = 'resolution'
-            elif bill_id.startswith('CA'):
-                type = 'contract'
-            elif bill_id.startswith('PR'):
-                type = 'resolution'
-            # don't collect these
-            elif (bill_id.startswith('GBM') or
-                  bill_id.startswith('HFA') or
-                  bill_id.startswith('IG')):
-                continue
+
+    def add_documents(self,attachment_path,bill,is_version=False):
+        global bill_versions
+        base_url = "http://lims.dccouncil.us/Download/" #nothing is actual links. we'll have to concatenate to get doc paths (documents are hiding in thrice-stringified json. eek.)
+        for a in attachment_path:
+            doc_type = a["Type"]
+            doc_name = a["Name"]
+            rel_path = a["RelativePath"]
+            if doc_type and doc_name and rel_path:  
+                doc_url = base_url+rel_path+"/"+doc_name
             else:
-                # will break if type isn't known
-                raise ValueError('unknown bill type: %s' % bill_id)
+                self.logger.warning("Bad link for document {}".format(doc_name))
+                return
+
+            mimetype = "application/pdf" if doc_name.endswith("pdf") else None
+
+            #figure out if it's a version from type/name
+            possible_version_types = ["SignedAct","Introduction","Enrollment","Engrossment"]
+            for vt in possible_version_types:
+                if vt.lower() in doc_name.lower():
+                    is_version = True
+                    doc_type = vt 
+
+            if "amendment" in doc_name.lower():
+                doc_type = "Amendment"
+
+            if is_version:
+                if doc_url in bill_versions:
+                    self.logger.warning("Version {} has been seen multiple times. Keeping first version seen".format(doc_url))
+                else:
+                    bill.add_version(doc_type,doc_url,mimetype=mimetype)
+                    bill_versions.append(doc_url)
+                continue
+                    
+
+            bill.add_document(doc_type,doc_url,mimetype=mimetype)
 
-            bill = Bill(session, 'upper', bill_id, title, type=type)
-            self.scrape_bill(bill)
+    def date_format(self,d):
+        #the time seems to be 00:00:00 all the time, so ditching it with split
+        return datetime.datetime.strptime(d.split()[0],"%Y/%m/%d")
```

Comment:
+                continue
    +
    +
    +            bill.add_document(doc_type,doc_url,mimetype=mimetype)
    
    -            bill = Bill(session, 'upper', bill_id, title, type=type)
    -            self.scrape_bill(bill)
    +    def date_format(self,d):
    +        #the time seems to be 00:00:00 all the time, so ditching it with split
    +        return datetime.datetime.strptime(d.split()[0],"%Y/%m/%d")